source/server/converters.h

```diff
@@ -317,6 +318,19 @@ typename TypeToStorageType<TDriverType, TGrpcType>::StorageType allocate_output_
 }
 
 }  // namespace converters
+
+const int kMaxGrpcErrorDescriptionSize = 4096;
+
+inline ::grpc::Status ApiErrorAndDescriptionToStatus(google::protobuf::int32 status, std::string& description)
+{
+    converters::trim_trailing_nulls(description);
+    // TODO: May need full-fledged json escape logic.
+    for (size_t index = 0; (index = description.find('"', index)) != std::string::npos; ++index) {
+        description.replace(index, 1, "\\\"");
+    }
+    std::string errorMessage = "{ \"code\":\"" + std::to_string(status) + "\", \"" + description + "\" }";
```

Comment:
Representing the error code as a number rather than a string would make it easier to parse. Error codes should not be more than 32 bits, so there is no problem with JSON implementations that use floating point internally limiting to 2^53.

Reply:
Thanks, With keys requiring quotes I went overboard.